NoteFind/Views/NoteCell.m

```diff
@@ -24,9 +35,33 @@ - (void)setNote:(Note *)note {
     _note = note;
     _postTitle.text = _note.title;
     _postCaption.text = _note.noteDescription;
-    self.postUsername.text = _note.author.username;
+    _postUsername.text = _note.author.username;
     _postNote.file = _note.note;
     [_postNote loadInBackground];
+    
+    // TODO: add addCount
+}
+- (IBAction)didTapDownload:(id)sender {
+    NSManagedObject *newOfflineNote = [NSEntityDescription insertNewObjectForEntityForName:@"OfflineNote" inManagedObjectContext:context];
+    NSData *fileData = [_postNote.file getData];
+    if (fileData == nil) {
```

Comment:
How do we handle notes that we've already downloaded to make sure there aren't duplicate objects in CoreData?

Reply:
fixed- i made an offline data manager to handle all of that and moved my core data code to that file.